File: leetcode/q217.go
package leetcode

// 217. Contains Duplicates

// TIME: O(n)
// LeetCode: Runtime: 91ms - Memory: 10.1MB
func containsDuplicate(nums []int) bool {
	numDuplicateCheck := make(map[int]bool, 0)
	for _, num := range nums {
		if numDuplicateCheck[num] {
			return true
		}
		numDuplicateCheck[num] = true
	}
	return false
}

File: src/q217.rs
use std::{collections::HashSet};

pub struct Solution {}
impl Solution {
    // LEETCODE: RunTime: 10ms - Memory: 3MB
    // TIME: O(n) - SPACE: O(n)
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut set = HashSet::new();
        for i in nums {
            if set.contains(&i) {
                return true;
            }
            set.insert(i);
        }
        false
    }
    // LEETCODE: RunTime: 23ms - Memory: 3.5MB
    pub fn contains_duplicate_v2(nums: Vec<i32>) -> bool {
        let set: HashSet<i32> = nums.clone().into_iter().collect();
        return set.len() != nums.clone().len();
    }
}
